components/region_cache_memory_engine/src/background.rs

```diff
@@ -242,6 +300,35 @@ impl Runnable for BackgroundRunner {
                 }
                 self.gc_finished();
             }
+            BackgroundTask::LoadTask => {
+                let skiplist_engine = {
+                    let core = self.engine_core.read().unwrap();
+                    core.engine().clone()
+                };
+                while let Some((range, snap)) = self.get_range_to_load() {
+                    let iter_opt = IterOptions::new(
+                        Some(KeyBuilder::from_vec(range.start.clone(), 0, 0)),
+                        Some(KeyBuilder::from_vec(range.end.clone(), 0, 0)),
+                        false,
+                    );
+                    for &cf in DATA_CFS {
+                        let handle = skiplist_engine.cf_handle(cf);
+                        match snap.iterator_opt(cf, iter_opt.clone()) {
+                            Ok(mut iter) => {
+                                iter.seek_to_first().unwrap();
+                                while iter.valid().unwrap() {
+                                    // use 0 sequence number here as the kv is clearly visible
+                                    let encoded_key = encode_key(iter.key(), 0, ValueType::Value);
+                                    handle.put(encoded_key, iter.value().to_vec());
+                                    iter.next().unwrap();
+                                }
+                            }
+                            Err(e) => {}
+                        }
+                    }
+                    self.on_snapshot_loaded(range).unwrap();
```

Comment:
is this only going to fail in exceptional cases?

Reply:
Yes, it should not be failed.